src/Apps/Legion/Components/CookSource/CookSource.types.ts

```diff
@@ -0,0 +1,17 @@
+import { IStyle } from '@fluentui/react';
+import { IADXAdapterTargetContext } from '../../Models/Interfaces';
+import { SourceType } from '../DataPusher/DataPusher.types';
+import { ITable } from '../../Adapters/Standalone/DataManagement/Models/DataManagementAdapter.types';
+import { ICookSource } from '../../Models';
+
+export interface ICookSourceProps {
+    onSourceTypeChange: (sourceType: SourceType) => void;
+    onSourceChange: (source: ICookSource) => void;
+    onGetTableData?: (table: ITable) => void;
```

Comment:
why does the parent get notified when it fetches data?

Reply:
I need table information in some cases (e.g. to show table content in data pusher after selecting table)